bootstrapper/internal/kubernetes/kubernetes.go

```diff
@@ -237,11 +257,23 @@ func (k *KubeWrapper) JoinCluster(ctx context.Context, args *kubeadm.BootstrapTo
 	if err != nil {
 		return fmt.Errorf("encoding kubeadm join configuration as YAML: %w", err)
 	}
-	log.With(slog.String("apiServerEndpoint", args.APIServerEndpoint)).Info("Joining Kubernetes cluster")
-	if err := k.clusterUtil.JoinCluster(ctx, joinConfigYAML, log); err != nil {
+
+	k.log.With(slog.String("apiServerEndpoint", args.APIServerEndpoint)).Info("Joining Kubernetes cluster")
+	if err := k.clusterUtil.JoinCluster(ctx, joinConfigYAML, k.log); err != nil {
 		return fmt.Errorf("joining cluster: %v; %w ", string(joinConfigYAML), err)
 	}
 
+	// If on control plane (and thus with etcd), try to prioritize etcd I/O.
+	if peerRole == role.ControlPlane {
+		k.log.Info("Prioritizing etcd I/O")
+		err = k.etcdIOPrioritizer.PrioritizeIO()
+		if errors.Is(err, etcdio.ErrNoEtcdProcess) {
+			k.log.Warn("No etcd process found, skipping I/O prioritization. Is this a single-node cluster?")
+		} else if err != nil {
+			return fmt.Errorf("prioritizing etcd I/O: %w", err)
```

Comment:
Do we really want to fail hard in this case? The way we detect etcd processes is inherently racy, so it might be better to just warn in that case.

Reply:
That's a good point.